sql/src/planner.rs

```diff
@@ -380,6 +387,15 @@ impl<'a, P: MetaProvider> PlannerDelegate<'a, P> {
 
         // Timestamp column must be provided.
         let timestamp_name = timestamp_name.context(RequireTimestamp)?;
+        // The timestamp key column must not be a Tag column
+        if let Some(timestamp_column) = name_column_map.get(&timestamp_name as &str) {
+            ensure!(
+                !timestamp_column.is_tag,
+                TimestampKeyTag {
+                    name: &timestamp_name as &str,
+                }
+            )
+        }
```

Comment:
I guess `as &str` is no need:
```suggestion
        if let Some(timestamp_column) = name_column_map.get(&timestamp_name) {
            ensure!(
                !timestamp_column.is_tag,
                TimestampKeyTag {
                    name: &timestamp_name,
                }
            )
        }
```

Reply:
if I remove here
 if let Some(timestamp_column) = name_column_map.get(&timestamp_name as &str)

error[E0277]: the trait bound `&str: Borrow<std::string::String>` is not satisfied
   --> sql/src/planner.rs:391:57
    |
391 |         if let Some(timestamp_column) = name_column_map.get(&timestamp_name) {
    |                                                         ^^^ the trait `Borrow<std::string::String>` is not implemented for `&str`
    |
    = help: the trait `Borrow<str>` is implemented for `std::string::String`
note: required by a bound in `BTreeMap::<K, V, A>::get`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `sql` due to previous error

But we can remove here

 TimestampKeyTag {
                    name: &timestamp_name,
                }

Reply:
I see. I guess we can put it as:
```rust
        if let Some(timestamp_column) = name_column_map.get(&timestamp_name.as_str()) {
            ensure!(
                !timestamp_column.is_tag,
                TimestampKeyTag {
                    name: &timestamp_name,
                }
            )
        }
```